src/components/resilience_component.cpp

```diff
@@ -128,126 +125,100 @@ std::size_t resilience::protectee_rank_ = 0;
         failure_detection_loop_async();
     }
 
-    void resilience::global_w_exec_finish_wrapper(work_item const& w)
-    {
-        allscale::resilience::get().w_exec_finish_wrapper(w);
-    }
-
-    // equiv. to taskAcquired in prototype
-    void resilience::w_exec_start_wrapper(work_item const& w) {
-        if (get_running_ranks() < 2 || env_resilience_disabled) 
-            return;
-
-        if (w.id().depth() != get_cp_granularity()) return;
-
-        //@ToDo: do I really need to block (via get) here?
-        if (get_running_ranks() > 1) {
-            hpx::async<remote_backup_action>(guard_, w).get();
-	    std::unique_lock<mutex_type> lock(backup_mutex_);
-	    //thread_safe_printer("local_backups_["+w.id().name()+"]\n");
-	    local_backups_[w.id().name()] = w;
+    void resilience::reschedule_dispatched_to_dead(size_t dead_rank, size_t token) {
+        std::multimap<size_t, work_item > delegated_items_copy;
+        {
+            std::unique_lock<mutex_type> lock(delegated_items_mutex_);
+            std::swap(delegated_items_, delegated_items_copy);
```

Comment:
I think, instead of swapping the whole map, we should copy and erase the ones from the dead rank only.